src/Client/Clients/TextDocumentClient.Completions.cs

```diff
@@ -0,0 +1,64 @@
+﻿using OmniSharp.Extensions.LanguageServer.Models;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OmniSharp.Extensions.LanguageServerProtocol.Client.Clients
+{
+    using Utilities;
+
+    /// <summary>
+    ///     Client for the LSP Text Document API.
+    /// </summary>
+    public partial class TextDocumentClient
+    {
+        /// <summary>
+        ///     Request completions at the specified document position.
+        /// </summary>
+        /// <param name="filePath">
+        ///     The full file-system path of the text document.
+        /// </param>
+        /// <param name="line">
+        ///     The target line (0-based).
+        /// </param>
+        /// <param name="column">
+        ///     The target column (0-based).
+        /// </param>
+        /// <param name="cancellationToken">
+        ///     An optional <see cref="CancellationToken"/> that can be used to cancel the request.
+        /// </param>
+        /// <returns>
+        ///     A <see cref="Task{TResult}"/> that resolves to the completions or <c>null</c> if no completions are available at the specified position.
+        /// </returns>
+        public Task<CompletionList> Completions(string filePath, int line, int column, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (String.IsNullOrWhiteSpace(filePath))
+                throw new ArgumentException($"Argument cannot be null, empty, or entirely composed of whitespace: {nameof(filePath)}.", nameof(filePath));
+
+            return PositionalRequest<CompletionList>("textDocument/completion", filePath, line, column, cancellationToken);
```

Comment:
_future_: Now that the strings are showing up in more places we should probably add a static constants class somewhere.  I'll add an issue for this.

Reply:
Yeah, I probably should have done this in the first place (even when my project was stand-alone) but haven't got around to it.